ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala

```diff
@@ -111,7 +111,8 @@ final class RecoveringIndexerSpec
       val resource = recoveringIndexer.start(() => testIndexer.subscribe())
       resource.asFuture
         .map { complete =>
-          readLog() should contain theSameElementsInOrderAs Seq(
+          // at this point the read log should at least contain logs informing about a successful indexer server startup
+          readLog().take(2) should contain theSameElementsInOrderAs Seq(
```

Comment:
if you flatten over `complete`, you can confidently assert the full output with `readLog()` (including the `(INFO,Successfully finished processing state updates)` line that is making it flaky now, right?

Reply:
Yes, but the next section handles that. This could be split into a separate test, I guess, named `should "return control once the indexer has started"` or something like that.

Reply:
Thank you for comments.
Should I add the additional test case in this PR?

Reply:
If you agree, sure. If not, I already approved. :slightly_smiling_face:

Reply:
Let's leave it in the current form as I saw some intentions to refactor the `RecoveringIndexer`. :)